packages/sdk/src/index.ts

```diff
@@ -158,13 +162,23 @@ export default class SDK {
         return reject(`Unexpected code flow - should never be here`);
       }
 
+      // decide which provider to use
+      let provider = reqOps.provider;
```

Comment:
add this point reqOps will hold the current values because of the merge at the beginning of the function.

so the logic can be as simple as this:

```
const provider = (reqOps.enableMev && req.method === 'eth_sendRawTransaction') ? reqOps.mevProtectionProvider : reqOps.provider;
```